### ISSUE TITLE
EditProfile Refinements

### ISSUE THREAD

**Issue by mikhael28** [mikhael28/paretOS#13]
Due to some fast and loose decision making in the past, the Edit Profile component has a lot of functionality/partial functionality that is there, bloating the file to a quite ridiculous 535 lines. We need to go through the component, see what state can be trimmed down and what additional refinements can be made.
1. saving the default language when the user makes a selection and showing a loading indicator - instead of relying on an ugly, mis-styled button.
2. Ability to edit/update GitHub username
3. Loading indicator when creating Projects
4. Editing your projects (and viewing them somewhere, in another component)

**Comment by nikhil12377** [comment]
i'd like to work on this.
but i'll need some guidance, i'm new to open source actually.

**Comment by mikhael28** [comment]
There are a few requirements here for this ticket, let's start you with something simpler. This one can be your second ticket, after you have had some time to get more familiar with the code-base. Can you work on this ticket please? https://github.com/mikhael28/paretOS/issues/23 Leave a comment there and let's continue the conversation!

**Comment by antonio-b** [comment]
@mikhael28  I'm interested in this issue, would like some more information. I haven't used React but would be up for the challenge.

**Comment by mikhael28** [comment]
@antonio-b sold, to the gentleman from Arizona! Assigned to you :) Let me know if you have any questions.

**Comment by antonio-b** [comment]
ha thanks

**Comment by antonio-b** [comment]
@mikhael28  Can I get a  screenshot or location of the component? would be much appreciated. thanks

**Comment by mikhael28** [comment]
Go to the profile tab, on the bottom left of the left nav

**Comment by mikhael28** [comment]
What you want to do, is to use Command+Shift+F and search for where <EditProfile is mounted. Then, check in the `Routes.js` where that might be.

**Comment by antonio-b** [comment]
Yes, I did find it. I searched through the repo on github to find the location. Now I'm reading through the code and watching tutorial over react state to have a better understanding.  Thanks

**Comment by antonio-b** [comment]
@mikhael28  I just want to ask a question. I assume that the main issue here is that the component is too large. So the solution would be is to break up this Editprofile.js into multiple,smaller components for easier readability, right? I'm finding it hard to follow the code when it covers over 500 lines. Research states to have components not exceed 250 lines with 100 lines being the ideal. I wanted to ask questions before spending too much time going this route. Thanks

**Comment by mikhael28** [comment]
Well, the line number isn't that crazy - really, the functionality that's important are the bullet points. If you can find a way to make some of those components into their owns files, great - however, the core of the ticket is to address those bullet points. 

Do you think I should break them up, into one piece at a time?

**Comment by mikhael28** [comment]
You are right though, in general components shouldn't be more than 250. Technically, this is a container - so it would have a bit more leeway. Before i broke it down to 550 lines, it used to be more. Do you have any particular questions about how the state works in this container component?

**Comment by antonio-b** [comment]
Not yet I don't. Still trying to understand the code and functionality.

**Comment by antonio-b** [comment]
I will work on it tomorrow and then follow up with questions. Thanks

**Comment by mikhael28** [comment]
Hey @antonio-b good morning! I sympathize, looking at that component again - there was too much crud. I pushed a quick little commit cleaning up some more of the state, removing some unnecessary code, and writing some comments to help you make sense of a few things. Please sync your fork with the latest from the `main` branch in my repository. you can review the commit on GitHub here: https://github.com/mikhael28/paretOS/commit/127f44c4c40ca6db81a65d076f5c13b7fe83f966

**Comment by mikhael28** [comment]
Now it's only 456 lines of code lol

**Comment by antonio-b** [comment]
Got it, thanks.

**Comment by mikhael28** [comment]
I'm going to go ahead and close this issue, as part of the original intent was accomplished here and the component was recently updated to use Hook state, instead of a Class component. However, this component needs quite some work - and I think should be prioritized higher. Right now, it's all the way on the third page, meaning almost no one sees it.

@jayeclark can you write up a new ticket based on the current needs of the component when you have time? That way, we can get it in front of some eye balls!

**Comment by mikhael28** [comment]
@antonio-b as an FYI, we've started using Puppeteer as our testing framework of choice! How is life in Selenium land? Would you be interested in writing a few tests with Puppeteer to learn about some of the differences?

**Comment by antonio-b** [comment]
@mikhael28 hello I'm actually in the process of learning cypress right now. We are switching to cypress for some automation work. Once I get an understanding of cypress, I'm willing to test in the other framework if needed.

**Comment by mikhael28** [comment]
Hahaha we just deleted Cypress from our repository in favor of Puppeteer. Let me know when you want to learn your third testing framework 😆

**Comment by antonio-b** [comment]
@mikhael28 lol right. 😆
